File: ProductManagementApi/Models/Product.cs
public class Product {
    public string ProductId { get; set; }
    public required string ProductName { get; set; }
    public string? ProductDescription { get; set; }
    public required decimal Price { get; set; }
    public required int QuantityInStock { get; set; }
    public string? Category { get; set; }
    public string? SupplierName { get; set; }
    public DateTime DateAdded { get; set; }
    public DateTime ExpirationDate { get; set; }
}

File: product-management-app/src/app/models/product.ts
export class Product {
    productId: string;
    productName: string;
    productDescription: string;
    price: number;
    quantityInStock: number;
    category: string;
    supplierName: string;
    dateAdded: Date;
    expirationDate: Date;
  
    constructor(
      productId: string,
      productName: string,
      productDescription: string,
      price: number,
      quantityInStock: number,
      category: string,
      supplierName: string,
      dateAdded: Date,
      expirationDate: Date
    ) {
      this.productId = productId;
      this.productName = productName;
      this.productDescription = productDescription;
      this.price = price;
      this.quantityInStock = quantityInStock;
      this.category = category;
      this.supplierName = supplierName;
      this.dateAdded = dateAdded;
      this.expirationDate = expirationDate;
    }
  }